Modules/_threadmodule.c

```diff
@@ -2523,7 +2523,9 @@ _thread__get_name_impl(PyObject *module)
     }
 
 #ifdef __sun
-    return PyUnicode_DecodeUTF8(name, strlen(name), "surrogateescape");
+    // Decode Solaris/Illumos (e.g. OpenIndiana) thread names as ASCII
+    // since OpenIndiana only supports ASCII names.
```

Comment:
```suggestion
    // gh-138004: Decode Solaris/Illumos (e.g. OpenIndiana) thread names
    //  from ASCII, since OpenIndiana only supports ASCII names.
```